create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -86,6 +93,54 @@ private Reference generateReference(DocTp docTp) {
         return reference;
     }
 
+    private List<Contributor> generateContributors(DocTp docTp) {
+        List<Contributor> contributors = new ArrayList<>();
+        extractAuthorGroup(docTp).forEach(authorGroupTp -> contributors.addAll(generateContributorsFromAuthorGroup(authorGroupTp)));
+        return contributors;
+    }
+
+    private List<Contributor> generateContributorsFromAuthorGroup(AuthorGroupTp authorGroupTp){
+        List<Contributor> contributors = new ArrayList<>();
```

Comment:
It is very rarely the case that we need to use side-effects to parse a list of objects. The Stream Api and the Guava Stream API are very powerful. If you have a need to flatten a list or to perform some kind of action of a collection and you see you are going towards mixing streams and iterative methodologies, try to see if there is already a solution for that.

Reply:
Fixed